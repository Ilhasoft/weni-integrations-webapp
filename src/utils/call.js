import { createApp } from 'vue';
import Alert from '../components/Alert/Alert.vue';

export default {
  callAlert({ props, containerRef }) {
    const AlertComponent = createApp(Alert, {
      ...props,
      onClose: () => {
        instance.$el.remove();
      },
      created() {
        this.$emit(['close'], () => {
          instance.$el.remove();
        });
      },
    });
    const element = document.createElement('div');
    const instance = AlertComponent.mount(element);

    if (containerRef) {
      instance.$el.style.position = 'absolute';
      containerRef.appendChild(instance.$el);
    } else {
      document.body.appendChild(instance.$el);
    }
  },
};
