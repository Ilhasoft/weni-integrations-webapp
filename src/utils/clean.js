/* eslint-disable no-unused-vars */
// Recursively remove empty items from nested objects.
function removeEmpty(obj) {
  if (Array.isArray(obj)) {
    return obj.filter((element, i) => {
      if (element && typeof element === 'object') {
        return removeEmpty(element);
      }
      if (element) {
        return element;
      }
    });
  }
  return Object.fromEntries(
    Object.entries(obj)
      .filter(([_, v]) => v != null)
      .map(([k, v]) => [k, v === Object(v) ? removeEmpty(v) : v]),
  );
}

export default removeEmpty;
