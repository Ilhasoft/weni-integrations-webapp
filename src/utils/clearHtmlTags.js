function clearHtmlTags(content) {
  let div = document.createElement('div');
  div.innerHTML = content;
  return div.textContent || div.innerText;
}

export { clearHtmlTags };
