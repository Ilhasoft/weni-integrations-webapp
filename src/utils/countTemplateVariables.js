const variablesRegex = /{{[0-9]+}}/g;

function countVariables(text) {
  if (!text) {
    return 0;
  }

  return (text.match(variablesRegex) || []).length;
}

export { countVariables };
