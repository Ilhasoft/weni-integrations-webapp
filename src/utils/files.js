import axios from 'axios';

async function dataUrlToFile(dataUrl, fileName, ignoreQuery = false) {
  if (!dataUrl) {
    return null;
  }

  if (!ignoreQuery && !dataUrl.startsWith('data:')) {
    dataUrl = dataUrl.concat('?file');
  }

  const res = await axios.get(dataUrl, { responseType: 'blob' });
  const blob = await res.data;
  return new File([blob], fileName, { type: res.headers['content-type'] });
}

async function toBase64(fileUrl) {
  if (!fileUrl) {
    return null;
  }

  return new Promise((resolve, reject) => {
    const reader = new FileReader();
    const blob = new Blob([fileUrl], { type: fileUrl.type });
    reader.readAsDataURL(blob);
    reader.onload = () => resolve(reader.result);
    reader.onerror = (error) => reject(error);
  });
}

async function getHeightAndWidthFromDataUrl(dataURL) {
  return new Promise((resolve) => {
    const img = new Image();
    img.onload = () => {
      resolve({
        height: img.height,
        width: img.width,
      });
    };
    img.src = dataURL;
  });
}

export { dataUrlToFile, toBase64, getHeightAndWidthFromDataUrl };
