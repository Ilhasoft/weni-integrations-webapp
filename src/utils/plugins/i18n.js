import Vue from 'vue';
import VueI18n from 'vue-i18n';

Vue.use(VueI18n);

import en from '../../locales/en.json';
import pt_br from '../../locales/pt_br.json';
import es_es from '../../locales/es_es.json';

const languages = {
  'en-us': en,
  'pt-br': pt_br,
  es: es_es,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en-us',
  fallbackLocale: 'en-us',
  messages,
});

export default i18n;
