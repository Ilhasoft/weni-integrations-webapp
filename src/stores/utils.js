import { clearHtmlTags } from '@/utils/clearHtmlTags';

export function formatGenericApp(appList) {
  return appList.map((app) => {
    const summary =
      app.code === 'generic' ? clearHtmlTags(app.config.channel_claim_blurb) : app.summary;

    return { ...app, generic: app.code === 'generic', summary };
  });
}
