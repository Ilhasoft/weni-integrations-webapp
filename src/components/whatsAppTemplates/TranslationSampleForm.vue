<template>
  <div>
    <unnnic-modal
      class="sample-form"
      :text="$t('WhatsApp.templates.sample.title')"
      scheme="feedback-yellow"
      modal-icon="alert-circle-1"
      @close="closeSampleModal"
      @click.stop
    >
      <template #message>
        <div>
          <span>
            {{ $t('WhatsApp.templates.sample.description') }}
          </span>
          <div class="sample-form__container">
            <div class="sample-form__preview__title">
              {{ $t('WhatsApp.templates.sample.preview') }}
            </div>
            <TemplatePreview
              class="sample-form__preview"
              :showTitle="false"
              :bodyOverwrite="formattedBody"
              :headerOverwrite="fileToPreview"
            />

            <div class="sample-form__content">
              <div v-if="hasMedia" class="sample-form__header__wrapper">
                <span class="sample-form__header__title">
                  {{ $t('WhatsApp.templates.sample.header') }}
                </span>
                <unnnic-button type="secondary" @click="() => this.$refs.file.click()">
                  {{ $t('WhatsApp.templates.sample.choose_file') }}
                </unnnic-button>
                <input
                  type="file"
                  ref="file"
                  :accept="supportedFormats"
                  :multiple="false"
                  @input="handleFileChange"
                  style="display: none"
                />
                <span v-if="file" class="sample-form__header__file-name"> {{ file.name }} </span>
              </div>
              <div v-if="hasVariables" class="sample-form__body__wrapper">
                <span class="sample-form__body__title">
                  {{ $t('WhatsApp.templates.sample.body') }}</span
                >
                <unnnic-input
                  v-for="(variable, index) in variableCount"
                  :key="variable"
                  :placeholder="`Enter content for {{${variable}}}`"
                  @input="handleVariableChange(index, $event)"
                />
              </div>
            </div>

            <div class="sample-form__button">
              <unnnic-button type="secondary" @click="saveSample">
                {{ $t('WhatsApp.templates.sample.send') }}
              </unnnic-button>
            </div>
          </div>
          <div class="sample-form__info">{{ $t('WhatsApp.templates.sample.info') }}</div>
        </div>
      </template>
    </unnnic-modal>
  </div>
</template>

<script>
  import { mapState } from 'pinia';
  import { whatsapp_store } from '@/stores/modules/appType/channels/whatsapp.store';
  import { unnnicCallAlert } from '@weni/unnnic-system';
  import TemplatePreview from '@/components/whatsAppTemplates/TemplatePreview.vue';

  import { countVariables } from '@/utils/countTemplateVariables.js';
  import { toBase64 } from '@/utils/files.js';

  export default {
    name: 'TranslationSampleForm',
    components: { TemplatePreview },
    props: {
      hasMedia: {
        type: Boolean,
        default: false,
      },
      hasVariables: {
        type: Boolean,
        default: false,
      },
    },
    data() {
      return {
        variablesData: [],
        formattedBody: '',
        file: null,
        fileToPreview: null,
      };
    },
    created() {
      this.formattedBody = this.templateTranslationCurrentForm.body;
    },
    computed: {
      ...mapState(whatsapp_store, ['templateTranslationCurrentForm']),
      variableCount() {
        return countVariables(this.templateTranslationCurrentForm.body);
      },
      supportedFormats() {
        const formatsMap = {
          IMAGE: '.png,.jpg,.jpeg',
          VIDEO: '.mp4',
          DOCUMENT: '.pdf',
        };
        return formatsMap[this.templateTranslationCurrentForm.header.mediaType] || '';
      },
    },
    methods: {
      closeSampleModal() {
        this.$emit('close-modal');
      },
      handleVariableChange(index, event) {
        this.variablesData[index] = event;
        let newBody = this.templateTranslationCurrentForm.body;
        this.variablesData.forEach((variable, index) => {
          newBody = newBody.replaceAll(`{{${index + 1}}}`, variable);
        });
        this.formattedBody = newBody;
      },
      validFormat(files) {
        const formats = this.supportedFormats.replaceAll('.', '').split(',');
        const isValid = Array.from(files).find((file) => {
          const validFormat = formats.find((format) => {
            return file.type.toLowerCase().includes(format.toLowerCase());
          });
          return validFormat;
        });
        return isValid;
      },
      validSize(files) {
        const maxFileSize = 16; // 16Mb
        const isValid = Array.from(files).find((file) => {
          const sizeInMB = (file.size / (1024 * 1024)).toFixed(2);
          return sizeInMB < maxFileSize;
        });
        return isValid;
      },
      validateFiles(files) {
        if (!files.length || files.length !== 1) {
          return false;
        }

        if (!this.validFormat(files)) {
          this.callErrorModal({ text: this.$t('WhatsApp.templates.error.invalid_file_format') });
          return false;
        }
        if (!this.validSize(files)) {
          this.callErrorModal({ text: this.$t('WhatsApp.templates.error.invalid_file_size') });
          return false;
        }
        return true;
      },
      async handleFileChange(event) {
        const { files } = event.target;
        if (this.validateFiles(files)) {
          this.file = files[0];

          if (this.templateTranslationCurrentForm.header.mediaType === 'IMAGE') {
            this.fileToPreview = await toBase64(files[0]);
          } else {
            this.fileToPreview = null;
          }
        }
        /* istanbul ignore next */
        if (this.$refs.file) {
          this.$refs.file.value = '';
        }
      },
      async saveSample() {
        if (this.hasVariables && this.variablesData.length !== this.variableCount) {
          this.callErrorModal({
            text: this.$t('WhatsApp.templates.error.missing_variable_example'),
          });
          return;
        }

        if (this.hasMedia && this.file === null) {
          this.callErrorModal({ text: this.$t('WhatsApp.templates.error.missing_media_example') });
          return;
        }

        const fileData = await toBase64(this.file);
        const sampleForm = {
          variables: this.variablesData,
          headerFile: fileData,
        };

        this.$emit('sample-submission', sampleForm);
        this.$emit('close-modal');
      },
      callErrorModal({ text }) {
        unnnicCallAlert({
          props: {
            text,
            title: this.$t('general.error'),
            icon: 'check-circle-1-1',
            scheme: 'feedback-red',
            position: 'bottom-right',
            closeText: this.$t('general.Close'),
          },
          seconds: 6,
        });
      },
    },
  };
</script>

<style lang="scss" scoped>
  .sample-form {
    &__container {
      max-height: 50vh;
      overflow-x: hidden;
      padding: 0 16px;
      overflow-y: auto;
    }

    &__content {
      overflow: auto;
      margin: $unnnic-spacing-stack-sm 0;
      padding-right: $unnnic-spacing-inline-xs;
    }

    &__preview {
      max-height: 30vh;
      overflow: auto;

      &__title {
        margin-bottom: $unnnic-spacing-stack-xs;
        text-align: left;
      }

      margin: 0 (-$unnnic-spacing-inline-md);
      padding: $unnnic-spacing-stack-nano $unnnic-spacing-inline-md;

      ::v-deep .template-preview__content,
      ::v-deep .template-preview__buttons {
        margin: $unnnic-spacing-stack-xs 0;
      }

      ::v-deep .template-preview__content {
        width: 210px;
        margin-bottom: $unnnic-spacing-stack-nano;
      }

      ::v-deep .template-preview__buttons {
        width: 225px;
        margin-top: $unnnic-spacing-stack-nano;
      }
    }

    &__body,
    &__header {
      &__wrapper {
        display: flex;
        flex-direction: column;
        text-align: left;
        gap: $unnnic-spacing-stack-xs;
        width: 75%;
      }
    }

    &__header {
      &__wrapper {
        margin-bottom: $unnnic-spacing-stack-sm;
      }

      &__file-name {
        color: $unnnic-color-neutral-clean;
        font-size: $unnnic-font-size-body-gt;
        font-weight: $unnnic-font-weight-bold;
      }
    }

    &__button {
      text-align: right;
      margin-bottom: $unnnic-spacing-stack-sm;

      .unnnic-button {
        width: 50%;
      }
    }

    &__info {
      background-color: $unnnic-color-neutral-soft;
      margin: 0 (-$unnnic-spacing-inline-md);
      margin-bottom: (-$unnnic-spacing-stack-giant);
      padding: $unnnic-spacing-inline-md;
    }
  }
</style>
