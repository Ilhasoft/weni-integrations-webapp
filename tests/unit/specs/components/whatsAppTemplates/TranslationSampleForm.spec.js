import { unnnicCallAlert as mockUnnnicCallAlert } from '@weni/unnnic-system';

jest.mock('@weni/unnnic-system', () => ({
  ...jest.requireActual('@weni/unnnic-system'),
  unnnicCallAlert: jest.fn(),
}));

import { toBase64 as mockToBase64 } from '@/utils/files';
jest.mock('@/utils/files', () => ({
  ...jest.requireActual('@/utils/files'),
  toBase64: jest.fn(),
}));

import Vuex from 'vuex';
import { mount, createLocalVue } from '@vue/test-utils';
import TranslationSampleForm from '@/components/whatsAppTemplates/TranslationSampleForm.vue';
import { unnnicModal } from '@weni/unnnic-system';
import i18n from '@/utils/plugins/i18n';

const localVue = createLocalVue();
localVue.use(Vuex);

const mountComponent = ({
  hasMedia = false,
  hasVariables = false,
  header = {
    header_type: 'TEXT',
  },
  body = '',
} = {}) => {
  const getters = {
    templateTranslationCurrentForm: jest.fn(() => {
      return { header, body };
    }),
  };

  const store = new Vuex.Store({
    modules: {
      WhatsApp: {
        namespaced: true,
        getters,
      },
    },
  });

  const wrapper = mount(TranslationSampleForm, {
    localVue,
    store,
    i18n,
    propsData: {
      hasMedia,
      hasVariables,
    },
    stubs: {
      unnnicModal,
    },
  });

  return { wrapper, getters };
};

describe('components/whatsAppTemplates/TranslationSampleForm.vue', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should be rendered properly', () => {
    const { wrapper } = mountComponent();
    expect(wrapper).toMatchSnapshot();
  });

  describe('variableCount', () => {
    it('should return zero as variable count', () => {
      const body = 'No Variables';
      const { wrapper } = mountComponent({ body });

      expect(wrapper.vm.variableCount).toEqual(0);
    });

    it('should return correct variable count', () => {
      const body = 'Variable {{1}}';
      const { wrapper } = mountComponent({ body });

      expect(wrapper.vm.variableCount).toEqual(1);
    });
  });

  describe('supportedFormats', () => {
    it('should return valid IMAGE formats', () => {
      const header = { header_type: 'MEDIA', mediaType: 'IMAGE' };
      const { wrapper } = mountComponent({ header });

      expect(wrapper.vm.supportedFormats).toEqual('.png,.jpg,.jpeg');
    });

    it('should return valid VIDEO formats', () => {
      const header = { header_type: 'MEDIA', mediaType: 'VIDEO' };
      const { wrapper } = mountComponent({ header });

      expect(wrapper.vm.supportedFormats).toEqual('.mp4');
    });

    it('should return valid DOCUMENT formats', () => {
      const header = { header_type: 'MEDIA', mediaType: 'DOCUMENT' };
      const { wrapper } = mountComponent({ header });

      expect(wrapper.vm.supportedFormats).toEqual('.pdf');
    });

    it('should return empty string if not valid', () => {
      const header = { header_type: 'MEDIA', mediaType: 'UNKNOWN' };
      const { wrapper } = mountComponent({ header });

      expect(wrapper.vm.supportedFormats).toEqual('');
    });
  });

  describe('closeSampleModal()', () => {
    it('should emit close-modal', async () => {
      const { wrapper } = mountComponent();

      let event = wrapper.emitted('close-modal');
      expect(event).toBeFalsy();

      wrapper.vm.closeSampleModal();

      event = wrapper.emitted('close-modal');

      expect(event).toBeTruthy();
      expect(event.length).toBe(1);
    });
  });

  describe('handleVariableChange()', () => {
    it('should update variablesData', async () => {
      const { wrapper } = mountComponent();

      await wrapper.setData({ variablesData: ['data1', 'data2'] });

      wrapper.vm.handleVariableChange(1, 'new data 2');

      expect(wrapper.vm.variablesData).toEqual(['data1', 'new data 2']);
    });

    it('should update formatted body', () => {
      const body = 'Variable {{1}} {{2}}';
      const { wrapper } = mountComponent({ body });

      wrapper.vm.handleVariableChange(1, 'new text');

      expect(wrapper.vm.formattedBody).toEqual('Variable {{1}} new text');
    });
  });

  describe('validFormat()', () => {
    it('should return false if file type is incompatible with supported type', () => {
      const header = { header_type: 'MEDIA', mediaType: 'VIDEO' };
      const { wrapper } = mountComponent({ header });

      const files = [{ type: 'pdf' }];

      expect(wrapper.vm.validFormat(files)).toBeFalsy();
    });

    it('should return true if file type is compatible with supported type', () => {
      const header = { header_type: 'MEDIA', mediaType: 'VIDEO' };
      const { wrapper } = mountComponent({ header });

      const files = [{ type: 'mp4' }];

      expect(wrapper.vm.validFormat(files)).toBeTruthy();
    });
  });

  describe('validSize()', () => {
    it('should return false if file size is greater than 10Mb', () => {
      const header = { header_type: 'MEDIA', mediaType: 'VIDEO' };
      const { wrapper } = mountComponent({ header });

      const files = [{ size: 11000000 }];

      expect(wrapper.vm.validSize(files)).toBeFalsy();
    });

    it('should return true if file size lower than 10Mb', () => {
      const header = { header_type: 'MEDIA', mediaType: 'VIDEO' };
      const { wrapper } = mountComponent({ header });

      const files = [{ size: 9000000 }];

      expect(wrapper.vm.validSize(files)).toBeTruthy();
    });
  });

  describe('validateFiles()', () => {
    it('should return false if there are no files', () => {
      const { wrapper } = mountComponent();

      const files = [];

      expect(wrapper.vm.validateFiles(files)).toBeFalsy();
    });

    it('should return false if there more than one file', () => {
      const { wrapper } = mountComponent();

      const files = [{}, {}];

      expect(wrapper.vm.validateFiles(files)).toBeFalsy();
    });

    it('should return false if file type is not valid', () => {
      const header = { header_type: 'MEDIA', mediaType: 'VIDEO' };
      const { wrapper } = mountComponent({ header });

      const files = [{ type: 'pdf' }];

      expect(wrapper.vm.validateFiles(files)).toBeFalsy();
    });

    it('should return false if file size is not valid', () => {
      const header = { header_type: 'MEDIA', mediaType: 'VIDEO' };
      const { wrapper } = mountComponent({ header });

      const files = [{ type: 'mp4', size: 11000000 }];

      expect(wrapper.vm.validateFiles(files)).toBeFalsy();
    });

    it('should return true if file size and type are valid', () => {
      const header = { header_type: 'MEDIA', mediaType: 'VIDEO' };
      const { wrapper } = mountComponent({ header });

      const files = [{ type: 'mp4', size: 9000000 }];

      expect(wrapper.vm.validateFiles(files)).toBeTruthy();
    });
  });

  describe('handleFileChange()', () => {
    beforeEach(() => {
      jest.clearAllMocks();
    });

    it('should set file as event data if valid', async () => {
      const header = { header_type: 'MEDIA', mediaType: 'VIDEO' };
      const { wrapper } = mountComponent({ header });

      const file = { type: 'mp4', size: 9000000 };

      const event = {
        target: {
          files: [file],
        },
      };

      expect(wrapper.vm.file).toBe(null);
      await wrapper.vm.handleFileChange(event);
      expect(wrapper.vm.file).toEqual(file);
    });

    it('should not set file as event data if not valid', async () => {
      const header = { header_type: 'MEDIA', mediaType: 'VIDEO' };
      const { wrapper } = mountComponent({ header });

      const file = { type: 'pdf', size: 9000000 };

      const event = {
        target: {
          files: [file],
        },
      };

      expect(wrapper.vm.file).toBe(null);
      await wrapper.vm.handleFileChange(event);
      expect(wrapper.vm.file).toBe(null);
    });

    it('should set fileToPreview as base 64 result if mediaType is IMAGE', async () => {
      const header = { header_type: 'MEDIA', mediaType: 'IMAGE' };
      const { wrapper } = mountComponent({
        hasMedia: true,
        header,
      });
      mockToBase64.mockImplementationOnce(() => Promise.resolve('b64photo'));

      const file = { type: 'png', size: 900 };

      const event = {
        target: {
          files: [file],
        },
      };

      expect(wrapper.vm.fileToPreview).toBe(null);
      await wrapper.vm.handleFileChange(event);
      expect(wrapper.vm.fileToPreview).toBe('b64photo');
    });
  });

  describe('saveSample()', () => {
    beforeEach(() => {
      jest.clearAllMocks();
    });

    it('should call error modal if variable is missing', async () => {
      const { wrapper } = mountComponent({
        hasVariables: true,
        body: 'Has 2 variables {{1}} {{2}}',
      });
      const spy = spyOn(wrapper.vm, 'callErrorModal');

      wrapper.vm.variablesData = ['first'];

      await wrapper.vm.saveSample();

      expect(spy).toHaveBeenCalledTimes(1);
      expect(spy).toHaveBeenCalledWith({ text: 'Missing variable example' });
    });

    it('should call error modal if media is missing', async () => {
      const { wrapper } = mountComponent({
        hasMedia: true,
        body: 'Simple body',
      });
      const spy = spyOn(wrapper.vm, 'callErrorModal');

      wrapper.vm.file = null;

      await wrapper.vm.saveSample();

      expect(spy).toHaveBeenCalledTimes(1);
      expect(spy).toHaveBeenCalledWith({ text: 'Missing media example' });
    });

    it('should be successfull if variables are provided', async () => {
      const { wrapper } = mountComponent({
        hasVariables: true,
        body: 'Has 2 variables {{1}} {{2}}',
      });
      const spy = spyOn(wrapper.vm, 'callErrorModal');

      wrapper.vm.variablesData = ['first', 'second'];

      await wrapper.vm.saveSample();

      expect(spy).not.toHaveBeenCalled();

      let event = wrapper.emitted('sample-submission');

      expect(event).toBeTruthy();
      expect(event.length).toBe(1);
      expect(event[0]).toEqual([
        {
          variables: ['first', 'second'],
          headerFile: undefined,
        },
      ]);

      event = wrapper.emitted('close-modal');

      expect(event).toBeTruthy();
      expect(event.length).toBe(1);
    });

    it('should be successfull if meida is provided', async () => {
      const { wrapper } = mountComponent({
        hasMedia: true,
        body: 'Simple body',
      });
      const spy = spyOn(wrapper.vm, 'callErrorModal');

      mockToBase64.mockImplementationOnce(() => Promise.resolve('b64photo'));
      wrapper.vm.file = true;

      await wrapper.vm.saveSample();

      expect(spy).not.toHaveBeenCalled();

      let event = wrapper.emitted('sample-submission');

      expect(event).toBeTruthy();
      expect(event.length).toBe(1);
      expect(event[0]).toEqual([
        {
          variables: [],
          headerFile: 'b64photo',
        },
      ]);

      event = wrapper.emitted('close-modal');

      expect(event).toBeTruthy();
      expect(event.length).toBe(1);
    });
  });

  describe('callErrorModal', () => {
    const { wrapper } = mountComponent();
    it('should call unnnicCallAlert', () => {
      expect(mockUnnnicCallAlert).not.toHaveBeenCalled();
      wrapper.vm.callErrorModal({ text: 'error text' });
      expect(mockUnnnicCallAlert).toHaveBeenCalledTimes(1);
      expect(mockUnnnicCallAlert).toHaveBeenCalledWith(
        expect.objectContaining({
          props: {
            text: 'error text',
            title: expect.any(String),
            icon: expect.any(String),
            scheme: expect.any(String),
            position: expect.any(String),
            closeText: expect.any(String),
          },
          seconds: expect.any(Number),
        }),
      );
    });
  });
});
