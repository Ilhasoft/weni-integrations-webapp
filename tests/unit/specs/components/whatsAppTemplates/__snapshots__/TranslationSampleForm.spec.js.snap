// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/whatsAppTemplates/TranslationSampleForm.vue should be rendered properly 1`] = `
<div>
  <transition-stub data-v-4c459454="" name="modal" class="sample-form">
    <div data-v-4c459454="" class="unnnic-modal">
      <div data-v-4c459454="" class="unnnic-modal-container">
        <div data-v-4c459454="" class="unnnic-modal-container-background">
          <div data-v-4c459454="" class="unnnic-modal-container-background-body">
            <div data-v-4c459454="" class="unnnic-modal-container-background-body-close_icon"><span data-v-59b073b4="" data-v-4c459454="" class="unnnic-icon unnnic-icon__size--sm unnnic--clickable unnnic-icon-scheme--neutral-dark"><svg viewBox="0 0 40 40" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M34.0592 34.9982C33.9361 34.9985 33.8142 34.9744 33.7004 34.9272C33.5867 34.88 33.4835 34.8107 33.3969 34.7232L19.9995 21.3271L6.60213 34.7232C6.51561 34.811 6.4125 34.8808 6.29879 34.9284C6.18507 34.976 6.06303 35.0005 5.93976 35.0005C5.81648 35.0005 5.69444 34.976 5.58073 34.9284C5.46701 34.8808 5.3639 34.811 5.27739 34.7232C5.10209 34.5471 5.00367 34.3087 5.00367 34.0602C5.00367 33.8117 5.10209 33.5734 5.27739 33.3972L18.6748 19.9999L5.27739 6.60374C5.09992 6.42753 5.00244 6.19132 5.00244 5.94137C5.00244 5.69142 5.09992 5.45522 5.27739 5.279C5.4536 5.10154 5.68981 5.00406 5.93976 5.00406C6.18971 5.00406 6.42591 5.10154 6.60213 5.279L19.9995 18.6751L33.3969 5.27775C33.4834 5.18994 33.5865 5.1202 33.7002 5.0726C33.8139 5.025 33.936 5.00049 34.0592 5.00049C34.1825 5.00049 34.3046 5.025 34.4183 5.0726C34.532 5.1202 34.6351 5.18994 34.7216 5.27775C34.8092 5.36454 34.8787 5.46782 34.9261 5.58161C34.9735 5.69541 34.9979 5.81747 34.9979 5.94075C34.9979 6.06403 34.9735 6.18609 34.9261 6.29988C34.8787 6.41367 34.8092 6.51695 34.7216 6.60374L21.3242 20.0011L34.7216 33.3985C34.8969 33.5746 34.9953 33.813 34.9953 34.0615C34.9953 34.31 34.8969 34.5484 34.7216 34.7245C34.6347 34.8115 34.5315 34.8805 34.4178 34.9275C34.3041 34.9745 34.1822 34.9985 34.0592 34.9982Z" fill="#3B414D" class="primary"></path></svg></span></div>
            <!---->
            <div data-v-4c459454="" class="unnnic-modal-container-background-body-alert_icon unnnic-card-scheme--feedback-yellow--icon"><span data-v-59b073b4="" data-v-4c459454="" class="unnnic-icon unnnic-icon__size--xl unnnic-icon-scheme--feedback-yellow"><svg viewBox="0 0 40 40" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M20 27.5005C20.7767 27.5005 21.4062 26.8709 21.4062 26.0942C21.4062 25.3176 20.7767 24.688 20 24.688C19.2233 24.688 18.5938 25.3176 18.5938 26.0942C18.5938 26.8709 19.2233 27.5005 20 27.5005Z" fill="#3B414D" class="primary"></path><path d="M20 22.813C19.7514 22.813 19.5129 22.7142 19.3371 22.5384C19.1613 22.3626 19.0625 22.1241 19.0625 21.8755V11.563C19.0625 11.3143 19.1613 11.0759 19.3371 10.9001C19.5129 10.7243 19.7514 10.6255 20 10.6255C20.2486 10.6255 20.4871 10.7243 20.6629 10.9001C20.8387 11.0759 20.9375 11.3143 20.9375 11.563V21.8755C20.9375 22.1241 20.8387 22.3626 20.6629 22.5384C20.4871 22.7142 20.2486 22.813 20 22.813Z" fill="#3B414D" class="primary"></path><path d="M20 35.0005C11.7287 35.0005 5 28.2717 5 20.0005C5 11.7292 11.7287 5.00049 20 5.00049C28.2712 5.00049 35 11.7292 35 20.0005C35 28.2717 28.2712 35.0005 20 35.0005ZM20 6.87549C12.7625 6.87549 6.875 12.763 6.875 20.0005C6.875 27.238 12.7625 33.1255 20 33.1255C27.2375 33.1255 33.125 27.238 33.125 20.0005C33.125 12.763 27.2375 6.87549 20 6.87549Z" fill="#3B414D" class="primary"></path></svg></span></div>
            <div data-v-4c459454="" class="unnnic-modal-container-background-body-title"> Add Sample Content </div>
          </div>
          <div data-v-4c459454="" class="unnnic-modal-container-background-body-description-container">
            <div data-v-4c459454="" class="unnnic-modal-container-background-body-description">
              <div><span>
        You have the option to provide specific content examples for your template.
      </span>
                <div class="sample-form__container">
                  <div class="sample-form__preview__title">
                    Preview
                  </div>
                  <div class="template-preview sample-form__preview">
                    <!---->
                    <div class="template-preview__content">
                      <!---->
                      <!---->
                      <!---->
                      <!---->
                    </div>
                    <!---->
                  </div>
                  <div class="sample-form__content">
                    <!---->
                    <!---->
                  </div>
                  <div class="sample-form__button"><button data-v-bcdcecac="" class="unnnic-button unnnic-button--size-large unnnic-button--secondary unnnic-button-scheme--">
                      <!---->
                      <!---->
                      <!----><span data-v-bcdcecac="" class="unnnic-button__label">
            Send
            </span>
                      <!---->
                    </button></div>
                </div>
                <div class="sample-form__info">Make sure not to include any actual user or customer information, and provide only sample content in your examples.</div>
              </div>
            </div>
          </div>
          <!---->
          <!---->
        </div>
      </div>
    </div>
  </transition-stub>
</div>
`;
